graphene/src/document.rs

```diff
@@ -98,8 +98,11 @@ impl Document {
 		layers.reduce(|a, b| &a[..a.iter().zip(b.iter()).take_while(|&(a, b)| a == b).count()]).unwrap_or_default()
 	}
 
-	pub fn folders<'a>(&'a self, layers: impl Iterator<Item = &'a [LayerId]>) -> impl Iterator<Item = &'a [LayerId]> {
-		layers.filter(|layer| self.is_folder(layer))
+	pub fn folders<'a, T: AsRef<[LayerId]>>(&'a self, layers: impl Iterator<Item = T> + 'a) -> impl Iterator<Item = T> + 'a
```

Comment:
`T: AsRef<[LayerId]>` probably should be omitted because it's already in `where` clause.

Reply:
yeah, sorry forgot that